src/server/frontend_wayland/wayland_surface_observer.h

```diff
@@ -64,7 +64,10 @@ class WaylandSurfaceObserver : public scene::NullSurfaceObserver
     void latest_client_size(geometry::Size window_size)
     {
         impl->window_size = window_size;
-        impl->requested_size = {};
+        if (auto& r = impl->requested_size; r && r.value() == window_size)
+        {
+            r.reset();
+        }
```

Comment:
Any chance for a regression test?

Reply:
We need tests around this, and @tarek-y-ismail tried to write one of them. But we're not there yet